ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableDataProvider.kt

```diff
@@ -0,0 +1,60 @@
+package com.revenuecat.purchases.ui.revenuecatui.data.processed
+
+import android.content.Context
+import com.revenuecat.purchases.Package
+import java.text.NumberFormat
+import java.util.Currency
+import java.util.Locale
+
+@Suppress("UnusedParameter", "FunctionOnlyReturningConstant")
+internal class VariableDataProvider(
+    private val applicationContext: Context,
```

Comment:
@tonidero that was already a good refactor. I'd recommend one follow up (feel free to do in a separate PR), to deal with the fact that you can't make this implement an interface:
Instead of having `Context` as a parameter, we can make still change this to our own interface, like `ApplicationContext`. Then make an implementation that takes a `Context`.

That would remove the dependency here and allow us to mock it all the way from the top

Reply:
But feel free to let me know if that's overkill. I see that we're passing it in `TestData`, so maybe it's not a big deal to pass this in tests and previews?

Reply:
This was done in #1257